### PR TITLE
Date input logic and UI improvements

### LINKED ISSUE #858: Better default value for another event date
Currently when adding a new date to calendar event on http://localhost:5800/calendar/new it is browser default value:

Image

It could be a nicer default if it instead used + 24 hours of old value. I saw some events with messed up day 2 clock since they forgot to update this time. Could either introduce state here or read the value of the previous input via ref.

Also, even the first date should probably default to :00 instead of random minutes

### LINKED ISSUE #1082: Improve Date Input value handling and not send "NaN"s
Prerequisites:

- [x] Merged: #1070

See this thread for details: https://github.com/Sendouc/sendou.ink/pull/1070#discussion_r1008885911
Maybe we can fix the hydration warning somehow at the same time (though the loading-in is expected)

### PR DESCRIPTION
**Pull request by remmycat**
Closes #858 (leftover task for setting hours of the initial date to :00)
Closes #1082

### CODE REVIEW

File: app/components/DateInput.tsx [hunk 1]
```diff
@@ -2,62 +2,68 @@ import { useIsMounted } from "~/hooks/useIsMounted";
...
 
+export interface DateInputProps
+  extends Omit<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    "defaultValue" | "min" | "max" | "onChange" | "value"
+  > {
+  defaultValue?: Date;
+  min?: Date;
+  max?: Date;
+  onChange?: (newDate: Date | null) => void;
```
Comment (Sendouc) [inline app/components/DateInput.tsx:13]: Nice 👍

File: app/routes/calendar/new.tsx [hunk 2]
```diff
@@ -315,80 +327,91 @@ function DatesInput() {
...
   const NEW_CALENDAR_EVENT_HOURS_OFFSET = 24;
 
+  const addDate = () =>
+    setDatesInputState((current) => {
+      // .reverse() is mutating, but map/filter returns a new array anyway.
+      const lastValidDate = current
+        .map((e) => e.date)
+        .filter(isDefined)
+        .reverse()
+        .at(0);
```
Comment (Sendouc) [inline app/routes/calendar/new.tsx:337]: https://caniuse.com/mdn-javascript_builtins_array_at

This is pretty recent addition so there is a risk of crashing I believe? Maybe we can just do [0]

File: app/routes/calendar/new.tsx [hunk 3]
```diff
@@ -344,64 +345,73 @@ function DatesInput() {
...
+                </div>
+              );
+            })}
+          </div>
+          {/* "Add" button */}
+          {datesCount < CALENDAR_EVENT.MAX_AMOUNT_OF_DATES && (
+            <Button tiny variant="outlined" onClick={addDate}>
+              {t("common:actions.add")}
+            </Button>
+          )}
```
Comment (Sendouc) [inline app/routes/calendar/new.tsx:409]: Nit: maybe after all we could just extract it as React component on same file?

```
{datesCount < CALENDAR_EVENT.MAX_AMOUNT_OF_DATES && (
            <AddButton {..propsHere} />
          )}
```

**Review by Sendouc — APPROVED** [review]
💯

Really like the UI improvements as well! Great job :)

**Comment by remmycat** [comment]
Implemented the nits before merging it 🙂

**Pull request merged**